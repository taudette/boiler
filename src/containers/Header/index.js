import React from 'react';
import '../../styles/components/_header.scss'

const Header = () => (
  <header>
    <h1>Header for react boiler</h1>
  </header>
)

export default Header
