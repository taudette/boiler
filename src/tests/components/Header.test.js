import React from 'react';
import { shallow } from 'enzyme';
import Header from '../../containers/Header';
import toJSON from 'enzyme-to-json';

//use shallow rendering for testing single components without their children

test('should render Header correctly', () => {
  const wrapper = shallow(<Header />);
  expect(toJSON(wrapper)).toMatchSnapshot(); //toJSON removes unnecesary info in snapshot


  expect(wrapper.find('h1').length).toBe(1);
  expect(wrapper.find('h1').text()).toBe('Header for react boiler');
  // const renderer = new ReactShallowRenderer();
  // renderer.render(<Header />);
  // expect(renderer.getRenderOutput()).toMatchSnapshot();
  //console.log(renderer.getRenderOutput());
})