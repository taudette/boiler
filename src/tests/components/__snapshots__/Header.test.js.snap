// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header>
  <h1>
    Header for react boiler
  </h1>
</header>
`;
